include/engine/routing_algorithms/routing_base_ch.hpp

```diff
@@ -350,63 +350,29 @@ void retrievePackedPathFromSingleHeap(const SearchEngineData<Algorithm>::QueryHe
 // && source_phantom.GetForwardWeightPlusOffset() > target_phantom.GetForwardWeightPlusOffset())
 // requires
 // a force loop, if the heaps have been initialized with positive offsets.
-void search(const datafacade::ContiguousInternalMemoryDataFacade<Algorithm> &facade,
+void search(SearchEngineData<Algorithm> &engine_working_data,
```

Comment:
The first point is about `SearchEngineData`:  some algorithms may need internally additional heaps and there is no possibility to tell how to get heaps without `SearchEngineData`. The proposal  is possible if [`SearchEngineData<Algorithm>`](https://github.com/Project-OSRM/osrm-backend/blob/e56d01b2e9deb8c22a501305e2e587ccd993353f/include/engine/routing_algorithms.hpp#L125) is moved in places where it is needed. `faced` argument exposes `Algorithm` and `SearchEngineData<Algorithm>` can be derived directly in `search` functions. It is doable in this PR.

The second point about `std::array`: there some places that already use [`std::vector`](https://github.com/Project-OSRM/osrm-backend/blob/e56d01b2e9deb8c22a501305e2e587ccd993353f/src/engine/routing_algorithms/routing_base_ch.cpp#L239-L240), so it is not clear how to generalize different approaches to pass initial points as arguments. It is ok to pass initial nodes directly in heaps, so the interface can be simpler
```
std::pair<EdgeWeight, std::vector<NodeID>>
search(GraphT& graph,
       HeapT&,
       HeapT&,
       Args... );
```